### PR TITLE
HPCC-13580 Support ELF format resource without binutils

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please check this still works on osx after refactoring.
@mckellyln please review.

File: common/dllserver/thorplugin.cpp [hunk 1]
```diff
@@ -245,6 +249,67 @@ static void secscan (bfd *file, sec_ptr sec, void *userParam)
...
+    if (hdr->e_ident[EI_CLASS] != ELFCLASS64)
+    {
+        DBGLOG("Failed to extract resource %s: Does not appear to be a ELF 64-bit binary", filename);
+        return false;
+    }
+
+    //Check that there is a symbol table for the sections.
+    if (hdr->e_shstrndx == SHN_UNDEF)
+    {
+        DBGLOG("Failed to extract resource %s: Does include a section symbol table", filename);
```
Comment (mckellyln) [inline common/dllserver/thorplugin.cpp]: Does not includes a section ...

File: common/dllserver/thorplugin.cpp [hunk 2]
```diff
@@ -245,6 +249,67 @@ static void secscan (bfd *file, sec_ptr sec, void *userParam)
...
+        const Elf64_Shdr & section = sectionHeaders[iSect];
+        const char * sectionName = symbolTable + section.sh_name;
+        if (streq(sectionName, sectname))
+        {
+            data.append(section.sh_size, start_addr + section.sh_offset);
+            return true;
+        }
+    }
+
+    DBGLOG("Failed to extract resource %s: Does include a matching entry", filename);
```
Comment (mckellyln) [inline common/dllserver/thorplugin.cpp]: Does not include ....

**Comment by AttilaVamos** [comment]
Automated Smoketest Build: success
HPCC Start: OK
HPCC Stop: OK

**Comment by AttilaVamos** [comment]
Automated Smoketest Build: success
HPCC Start: OK
HPCC Stop: OK

**Comment by ghalliday** [comment]
@richardkchapman please check + review

**Pull request merged**